code/modules/mob/living/living.dm

```diff
@@ -645,37 +645,37 @@ mob/living/carbon/human/verb/stopSliding()
 
 /mob/living/proc/dive()
 	var/client/C = src.client
-	var/speed = movement_delay()
 	resting = TRUE
 	var/_dir = C.true_dir
-	var/_hunger = (MOB_BASE_MAX_HUNGER - nutrition)
-	if(_hunger >= 250) //Will be shown on overlay as orange nutrition
-		to_chat(src, SPAN_WARNING("You weakly slump down!")) //You fall down because the rest still procs; a huge disadvantage
-		return
+
 	if(ishuman(src) && !weakened && (_dir))// If true_dir = 0(src isn't moving), doesn't proc.
-		livmomentum = 5 // Set momentum value as soon as possible for stopSliding to work better
-		nutrition -= 25
 		var/mob/living/carbon/human/H = src
+		
+		if(momentum_dir == _dir)
+			livmomentum = H.momentum_speed // Set momentum value as soon as possible for stopSliding to work better
 		var/range = 1 //checks for move intent; dive one tile further if on run intent
-		if (move_intent.flags & MOVE_INTENT_EXERTIVE)
-			range++
+
+		// Diving
 		to_chat(H, SPAN_NOTICE("You dive onwards!"))
 		pass_flags += PASSTABLE // Jump over them!
 		H.allow_spin = FALSE
-		var/is_jump = FALSE
 		if(istype(get_step(H, _dir), /turf/simulated/open))
-			is_jump = TRUE
-		H.throw_at(get_edge_target_turf(H, _dir), range + is_jump, 1)// "Diving"; if you dive over a table, your momentum is set to 0. If you dive over space, you are thrown a tile further.
+			range++
+		if(H.momentum_speed > 4)
+			range++
+		H.throw_at(get_edge_target_turf(H, _dir), range, 1) // If you dive over a table, your momentum is set to 0. If you dive over space, you are thrown a tile further.
```

Comment:
```suggestion
		H.throw_at(get_edge_target_turf(H, _dir), range, 1) // If you dive over a table, your momentum is set to 0. If you dive over space, you are thrown 1 tile further.
```
Just to reboot the CI with a commit. I wouldn't have make the suggestion otherwise